packages/e2e-tests/e2e-server.js

```diff
@@ -46,7 +46,33 @@ async function startedOnPort(serverProcess, port, timeout) {
 	});
 }
 
-export async function serve(root, isBuild, port) {
+async function buildWatchIdle(watchProcess, timeout) {
+	let id;
+	let stdoutListener;
+	const timerPromise = new Promise(
+		(_, reject) =>
+			(id = setTimeout(() => {
+				reject(`timeout for server start after ${timeout}`);
+			}, timeout))
+	);
+	const startedPromise = new Promise((resolve, reject) => {
+		stdoutListener = (data) => {
+			const str = data.toString();
+			const match = str.match(/built in \d+ms\./);
```

Comment:
initial build of `build --watch` logs "build in 123ms." at the end. wait for that to ensure we don't try to load the preview before that is done